### DISCUSSION TITLE
Ajout nouveau service sur les versions pro ?

### DISCUSSION THREAD

**Discussion by netprecis** [AmauriC/tarteaucitron.js#709] [Q&A]
Bonjour, 
Je suis surpris de ne trouver aucune question de cet ordre dans les différentes discussion donc je me lance en espérant que ce n'est pas une question déjà maintes fois répondues...
J'ai bien compris comment ajouter un service potentiel avec l'intégration manuelle du script, mais existe-t-il un moyen d'ajouter un service sur la version avec abonnement  sans modifier du coup les JS ?
J'ai un client qui utilise les services de leadfeeder.com et je ne vois pas ce service dans la liste.
Ce même client souhaiterait ajouter dans tarteaucitron un cookie maison.
Je précise que tarteaucitron sera intégré via GTM avec les déclaration de service dans GTM également.
Faut-il réintégrer les scripts de TAU dans le site ?

**Comment by AmauriC** [comment]
Il est possible de charger des services perso avec la v pro en les incluants dans une fonction js spéciale:

```
function tarteaucitronCustomPremium() {

   // définition du service
   tarteaucitron.services.servicePerso1 = {
      "key": "servicePerso1",
      "type": "analytic",
      "name": "Nom du service",
      "needConsent": true,
      "cookies": ['cookie1', 'cookie2'],
      "readmoreLink": "",
      "js": function () {
            // action quand le service est autorisé
      },
      "fallback": function () {
           // action quand le service est désactivé
      }
   };
   // chargement du service
   (tarteaucitron.job = tarteaucitron.job || []).push('servicePerso1');
}
```

**Reply by netprecis** [reply]
Top merci !

**Reply by Cnedellec** [reply]
Bonjour Amaury, 

J'ai créé ce code en suivant votre doc : 

<script>
function tarteaucitronCustomPremium() {

{

   // définition du service
   tarteaucitron.services.TimeOne = {
      "key": "TimeOne",
      "type": "ads",
      "name": "TimeOne",
      "needConsent": true,
      "cookies": ['tarteaucitron', 'timeOne'],
      "readmoreLink": "",
      "js": function () {
            // action quand le service est autorisé
      },
      "fallback": function () {
           // action quand le service est désactivé
      }


   };

// définition du service
   tarteaucitron.services.Remailme = {
      "key": "Remailme",
      "type": "ads",
      "name": "Remailme",
      "needConsent": true,
      "cookies": ['tarteaucitron', 'Remailme'],
      "readmoreLink": "",
      "js": function () {
            // action quand le service est autorisé
      },
      "fallback": function () {
           // action quand le service est désactivé
      }


   };

}
   // chargement du service
   (tarteaucitron.job = tarteaucitron.job || []).push('TimeOne');
   (tarteaucitron.job = tarteaucitron.job || []).push('Remailme');
}
  
 </script>



Les tags sont bien présents dans le panneau de gestion des cookies. Je rencontre un soucis de persistance de cookies en cas de refus, le bandeau se recharge alors à chaque page. 

Merci de votre aide

**Reply by AmauriC** [reply]
Il ne faut pas mettre le cookie 'tarteaucitron' dans la liste des cookies sinon le consentement est supprimé et le bandeau réapparait ; )

**Reply by Cnedellec** [reply]
Merci Amauri :)

**Comment by JMAILLY** [comment]
Bonjour,
Je me permets de rebondir sur le sujet, car je n'arrive pas à trouver comment  faire remonter les statistiques de l'ajout de ces nouveaux services premium, y a-t-il un moyen ou ce n'est pas possible actuellement sur la version pro ? Avec mes remerciements,

**Reply by AmauriC** [reply]
La v pro ne comptabilise les stats que pour les services inclus sur tac, pas les services perso. 

Je note quand même l'idée de pouvoir les inclure ; )
